dask/dataframe/tests/test_groupby.py

```diff
@@ -554,3 +556,169 @@ def test_groupby_multiprocessing():
     with dask.set_options(get=get):
         assert_eq(ddf.groupby('B').apply(lambda x: x),
                   df.groupby('B').apply(lambda x: x))
+
+
+def test_groupby_normalize_index():
+    full = pd.DataFrame({'a': [1, 2, 3, 4, 5, 6, 7, 8, 9],
+                         'b': [4, 5, 6, 3, 2, 1, 0, 0, 0]},
+                        index=[0, 1, 3, 5, 6, 8, 9, 9, 9])
+    d = dd.from_pandas(full, npartitions=3)
+
+    assert d.groupby('a').index == 'a'
+    assert d.groupby(d['a']).index == 'a'
+    assert d.groupby(d['a'] > 2).index._name == (d['a'] > 2)._name
+    assert d.groupby(['a', 'b']).index == ['a', 'b']
+
+    assert d.groupby([d['a'], d['b']]).index == ['a', 'b']
+    assert d.groupby([d['a'], 'b']).index == ['a', 'b']
+
+
+@pytest.mark.parametrize('spec', [
+    {'b': {'c': 'mean'}, 'c': {'a': 'max', 'a': 'min'}},
+    {'b': 'mean', 'c': ['min', 'max']},
+    {'b': np.sum, 'c': ['min', np.max, np.std, np.var]},
+    ['sum', 'mean', 'min', 'max', 'count', 'size', 'std', 'var'],
+    'var',
+])
+@pytest.mark.parametrize('split_every', [False, None])
+@pytest.mark.parametrize('grouper', [
+    lambda df: 'a',
+    lambda df: ['a', 'd'],
+    lambda df: [df['a'], df['d']],
+    lambda df: df['a'],
+    lambda df: df['a'] > 2,
+])
+def test_aggregate__examples(spec, split_every, grouper):
+    pdf = pd.DataFrame({'a': [1, 2, 3, 1, 1, 2, 4, 3, 7] * 10,
+                        'b': [4, 2, 7, 3, 3, 1, 1, 1, 2] * 10,
+                        'c': [0, 1, 2, 3, 4, 5, 6, 7, 8] * 10,
+                        'd': [3, 2, 1, 3, 2, 1, 2, 6, 4] * 10},
+                       columns=['c', 'b', 'a', 'd'])
+    ddf = dd.from_pandas(pdf, npartitions=10)
+
+    assert_eq(pdf.groupby(grouper(pdf)).agg(spec),
+              ddf.groupby(grouper(ddf)).agg(spec, split_every=split_every))
```

Comment:
Thanks. After your little nudge towards `pytest.mark.parametrize`, I became I fan of it :)